compiler/utils.go

```diff
@@ -103,12 +104,12 @@ func (fc *funcContext) Delayed(f func()) {
 }
 
 // CollectDCEDeps captures a list of Go objects (types, functions, etc.)
-// the code translated inside f() depends on. The returned list of identifiers
-// can be used in dead-code elimination.
+// the code translated inside f() depends on. Then sets those objects
+// as dependencies of the given dead-code elimination info.
 //
 // Note that calling CollectDCEDeps() inside another CollectDCEDeps() call is
 // not allowed.
-func (fc *funcContext) CollectDCEDeps(f func()) []string {
+func (fc *funcContext) CollectDCEDeps(dce *dce.Info, f func()) {
```

Comment:
Consider this an optional request, but I would move this method into the dce package as well.

Reply:
Moved to `dce.Collector`